src/main/java/net/earthcomputer/clientcommands/command/TicTacToeCommand.java

```diff
@@ -0,0 +1,280 @@
+                return this.board[0][2];
+            }
+            return null;
+        }
+
+        private enum Mark {
+            NOUGHT(Component.translatable("ticTacToeGame.noughts")),
+            CROSS(Component.translatable("ticTacToeGame.crosses"));
+
+            private final Component name;
+
+            Mark(Component name) {
+                this.name = name;
+            }
+
+            public Mark opposite() {
+                return this == NOUGHT ? CROSS : NOUGHT;
+            }
+        }
+    }
+
+    private static class TicTacToeGameScreen extends Screen {
+        private final TicTacToeGame game;
+
+        private static final ResourceLocation GRID_TEXTURE = new ResourceLocation("clientcommands", "textures/tic_tac_toe/grid.png");
+        private static final ResourceLocation MARKS_TEXTURE = new ResourceLocation("clientcommands", "textures/tic_tac_toe/marks.png");
+
+        private static final int gridSizeTexture = 512;
+        private static final int markSizeTexture = 152;
+
+        private static final int gridSize = 256;
+        private static final int cellSize = 80;
+        private static final int borderSize = 8;
+        private static final int markSize = 76;
+        private static final int padding = 2;
+
+        private TicTacToeGameScreen(TicTacToeGame game) {
+            super(Component.translatable("ticTacToeGame.title", game.opponent.getProfile().getName()));
+            this.game = game;
+        }
+
+        @Override
+        public void renderBackground(GuiGraphics guiGraphics, int mouseX, int mouseY, float partialTick) {
+            super.renderBackground(guiGraphics, mouseX, mouseY, partialTick);
+            int startX = (this.width - gridSize) / 2;
+            int startY = (this.height - gridSize) / 2;
+
+            guiGraphics.drawString(this.font, this.title, startX, startY - 20, 0xff_ffffff);
+            guiGraphics.drawString(this.font, Component.translatable("ticTacToeGame.playingWith", this.game.yourMarks.name), startX, startY - 10, 0xff_ffffff);
+
+            guiGraphics.blit(GRID_TEXTURE, startX, startY, gridSize, gridSize, 0, 0, gridSizeTexture, gridSizeTexture, gridSizeTexture, gridSizeTexture);
+            TicTacToeGame.Mark[][] board = this.game.board;
+
+            for (byte x = 0; x < 3; x++) {
+                for (byte y = 0; y < 3; y++) {
+                    TicTacToeGame.Mark mark = board[x][y];
+                    if (mark == null) {
+                        continue;
+                    }
+                    int offset = switch (mark) {
+                        case NOUGHT -> 0;
+                        case CROSS -> markSizeTexture;
+                    };
+                    guiGraphics.blit(MARKS_TEXTURE, startX + (cellSize + borderSize) * x + padding, startY + (cellSize + borderSize) * y + padding, markSize, markSize, offset, 0, markSizeTexture, markSizeTexture, 2 * markSizeTexture, markSizeTexture);
```

Comment:
Well I am already using constants.

Reply:
i mean adding
`static final MARKS_TEXTURE_WIDTH`
`static final MARKS_TEXTURE_HEIGHT`
so that you won't propogate if you change the texture dimensions to add more textures to `marks.png`

Reply:
sorry, was unclear

Reply:
I think it's fine, there will only ever be two textures (noughts and crosses, no more), which is a small enough number to be hardcoded.